identity-apps-core/components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/util/AppPortalUtils.java

```diff
@@ -395,6 +395,7 @@ private static void shareApplication(String tenantDomain, int tenantId, String a
             PrivilegedCarbonContext privilegedCarbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
             privilegedCarbonContext.setTenantId(tenantId);
             privilegedCarbonContext.setTenantDomain(tenantDomain);
+            privilegedCarbonContext.setUsername(appOwner);
```

Comment:
Verified that tenant admin user's name is passed as the app owner.
![Screenshot 2024-02-15 at 14 03 09](https://github.com/wso2/identity-apps/assets/35717390/25a01c12-c72d-4d30-bc8f-41da0c8c6028)